src/Parallel/Algorithm.hpp

```diff
@@ -224,11 +224,22 @@ class AlgorithmImpl<ParallelComponent, tmpl::list<PhaseDepActionListsPack...>>
     p | inboxes_;
     p | array_index_;
     p | global_cache_proxy_;
-    // note that `perform_registration_or_deregistration` passes the `box_` by
+    // Note that `perform_registration_or_deregistration` passes the `box_` by
     // const reference. If mutable access is required to the box, this function
     // call needs to be carefully considered with respect to the `p | box_` call
     // in both packing and unpacking scenarios.
-    perform_registration_or_deregistration(p, box_);
+    //
+    // Note also that we don't perform (de)registrations when pup'ing for a
```

Comment:
is there same way to enforce this, e.g. by puping the global core ID and checking on restore that it matches what we expect? This might not work until we have support for checking if we are LB or CP

Reply:
Can you clarify? Are you suggesting we check each chare is restored onto the same proc# that it was first running on, as a way of checking the registration info is still valid?

I could probably do something like what I did in main, and if the phase is WriteCheckpoint then I add a code ID into the pup stream...? It's a bit hacky but might work?

Reply:
Yes, that's exactly what I was suggesting, but I'm not sure that that's even true. I saw your check in `Main` that the number of cores/nodes matches and I would guess that'll cover all cases. 

Exactly what you described is what I had in mind, but it just might not be a good idea :) I'm not sure if Charm++ guarantees exact 1-to-1 restore on restart

Reply:
Yeah I have no idea about the 1-to-1 restore question... I'm happy to put in checks and explore, or we go as-is for now. Let me know what you think :)

Reply:
let's go as-is for now :) Things seem to be working :)